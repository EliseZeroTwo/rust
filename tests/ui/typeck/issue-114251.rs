// compile-flags: -C debug-assertions

macro_rules! feedu8 {
    () => {
        0u8
    }
}

fn main() {
    let kitty = {
        feedu8!();
        //~^ HELP: remove this semicolon
    };

    let cat: u8 = kitty;
    //~^ ERROR: mismatched types
}
