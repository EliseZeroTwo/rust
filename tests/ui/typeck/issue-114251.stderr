error[E0308]: mismatched types
  --> $DIR/issue-114251.rs:15:19
   |
LL |         feedu8!();
   |                  - help: remove this semicolon
...
LL |     let cat: u8 = kitty;
   |              --   ^^^^^ expected `u8`, found `()`
   |              |
   |              expected due to this

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
